src/Device/jprogOperations.ts

```diff
@@ -183,17 +183,17 @@ export async function programFirmware(
 ) {
     try {
         const confirmed = await confirmHelper(deviceSetupConfig.promiseConfirm);
-        if (!confirmed) return undefined;
-
-        logger.debug(`Programming ${device.serialNumber} with ${fw}`);
-        await program(device.id, fw);
-        logger.debug(`Resetting ${device.serialNumber}`);
-        await reset(device.id);
+        if (confirmed) {
+            logger.debug(`Programming ${device.serialNumber} with ${fw}`);
+            await program(device.id, fw);
+            logger.debug(`Resetting ${device.serialNumber}`);
+            await reset(device.id);
+        }
+        return device;
```

Comment:
Revert/update this to early return `device` if the user declines and move the return of the device to after the `try/catch`.